extism.go

```diff
@@ -338,9 +338,9 @@ func NewPlugin(
 	//  - If there is only one module in the manifest then that is the main module by default
 	//  - Otherwise the last module listed is the main module
 
-	logLevel := Warn
-	if config.LogLevel != nil {
-		logLevel = *config.LogLevel
+	logLevel := LogLevelWarn
+	if config.LogLevel != LogLevelOff {
```

Comment:
As you mentioned, this is the only place where I am unsure. Because according to this logic the user can't set log level to `LogLevelOff`. But I like your suggestion of having a `LogLevelUnset` as iota

Reply:
We can add and unset variant or I thought of one more way, which is to revert the pointer field and then add a new package function for each level which returns a reference to the level e.g `extism.LogLevelInfo()` 

This would let us use `nil` again, haha I'm not sure which way is worse.